generator/src/main/kotlin/io/github/mscheong01/krotodc/util/StringExtensions.kt

```diff
@@ -0,0 +1,22 @@
+package io.github.mscheong01.krotodc.util
+
+fun String.snakeToCamelCaseNoRegex(): String {
+    var preChar = '.' // tricks, because the first char could not be dot(.) in real case.
+    return this.fold(StringBuilder()) { acc, s ->
+        acc.let {
+            if (s == '_') {
+                return@let acc
+            } else if (preChar == '.') {
+                acc.append(s)
+            } else if (s.isDigit()) {
+                acc.append(s)
+            } else if (preChar == '_' || preChar.isDigit()) {
+                acc.append(s.uppercase())
+            } else {
+                acc.append(s)
+            }
+        }
+        preChar = s
+        acc
+    }.toString()
+}
```

Comment:
Did you reference this from protobuf-java? It would be nice if you could add a github permalink to the referenced code

Reply:
This is my own implementation, aiming to replicate what I believe the protobuf-java library does. I will make an effort to locate the original implementation, but I cannot guarantee its completion.

Reply:
After a quick search, I thought the protoc builtin java plugin deal with the field name in "[UnderscoresToCamelCase](https://github.com/protocolbuffers/protobuf/blob/4cc258a6a1cb6d3e526e3ca2f6ab7748ee84e2e3/src/google/protobuf/compiler/cpp/helpers.cc#L292)" method (in C++).